qgsquick/from_qgis/attributes/qgsquickattributemodel.cpp

```diff
@@ -406,7 +406,14 @@ void QgsQuickAttributeModel::create()
 
 bool QgsQuickAttributeModel::hasAnyChanges()
 {
-  return FID_IS_NULL( mFeatureLayerPair.feature().id() ) || mFeatureLayerPair.layer()->editBuffer()->isFeatureAttributesChanged( mFeatureLayerPair.feature().id() );
+  if ( FID_IS_NULL( mFeatureLayerPair.feature().id() ) ) return true;
```

Comment:
Can `mFeatureLayerPair.feature()` be also null?

Reply:
Its not returning a pointer, but an const instance. So this should be at least safe to call as well as additional call for its id.

However, it can return "empty" feature, which should have invalid id. Maybe the question can be if we want to also check mFeatureLayerPair for validity (checking layer, feature geometry and feature.isValid) or only check feature validity. However, I think its not needed in this function.